mongorestore/mongorestore.go

```diff
@@ -215,22 +214,11 @@ func (restore *MongoRestore) ParseAndValidateOptions() error {
 
 	// deprecations with --nsInclude --nsExclude
 	if restore.NSOptions.DB != "" || restore.NSOptions.Collection != "" {
-		// log a deprecation message if restoring from a directory, or
-		// from a non-bson file with the --archive option
-		_, fileType, err := restore.getInfoFromFile(restore.TargetDirectory)
-		if err != nil {
-			return err
-		}
-
-		isDir := false
-		if target, err := newActualPath(restore.TargetDirectory); err == nil {
-			isDir = target.IsDir()
-		}
-
-		if isDir || (fileType != BSONFileType && restore.InputOptions.Archive != "") {
-			log.Logvf(log.Always, "the --db and --collection args should only be used when "+
-				"restoring from a BSON file. Other uses are deprecated and will not exist "+
-				"in the future; use --nsInclude instead")
+		if filepath.Ext(restore.TargetDirectory) != ".bson" {
+			log.Logvf(log.Always,
+				`The --db and --collection flags are deprecated for 
+                                 this use-case; please use --nsInclude instead, 
+                                 i.e. with --nsInclude=${DATABASE}.${COLLECTION}`)
```

Comment:
The use of backticks here for the string literal means that the log message will print with all of the spaces. i.e., it'll print out like this:
```
$ mongorestore --db d --coll c dir/
The --db and --collection flags are deprecated for
                            this use-case; please use --nsInclude instead,
                            i.e. with --nsInclude=${DATABASE}.${COOLECTION}
```

I suggest using the old method of multi-line strings: double-quotes and breaking it up across newlines with `+`.

Reply:
Great catch! Thanks a lot for noticing this! I'll push a fix before merging.